---
title: Machine Learning Deriven Set of microRNAs as a Novel Biomarker for Myocardial Infarction Diagnosis
#titlerunning: Short form of title (if too long for head)
authorrunning: M. Samadishadlou et al.
#thanks: | 
#    Grants or other notes about the article that should go on the front 
#    page should be placed here. General acknowledgments should be placed at the
#    end of the article.

authors: 
- name: Mehrdad Samadishadlou
  address: Medical Nanotechnology gruop, Tabriz University of Medical Sciences, Tabriz, Iran
  email: samadishadlou@tbzmed.ac.ir
  
- name: Reza Rahbarghazi
  address: Medical Nanotechnology gruop, Tabriz University of Medical Sciences, Tabriz, Iran
  email: rahbarghazir@tbzmed.ac.ir
  
- name: Farhad Bani
  address: Medical Nanotechnology gruop, Tabriz University of Medical Sciences, Tabriz, Iran
  email: bainf@tbzmed.ac.ir

keywords:
- microRNA
- Machine Learning
- Myocardial Infarction
- Bioinformatics

#PACS: 
#- PAC1
#- superPAC
    
#MSC:
#- MSC code 1
#- MSC code 2

always_allow_html: true

abstract: |
  MicroRNAs (miRNAs) play a crucial role in regulating adaptive and maladaptive responses in cardiovascular diseases, making them attractive targets for potential therapeutics. However, their potential as novel biomarkers for diagnosing cardiovascular diseases requires systematic evaluation. In this study, we aimed to identify a key set of miRNA biomarkers using integrated bioinformatics and machine learning analysis. We combined and analyzed three gene expression datasets from the Gene Expression Omnibus (GEO) database, which contains peripheral blood mononuclear cell samples from individuals with myocardial infarction (MI), stable coronary artery disease (CAD), and healthy individuals. Additionally, we selected a set of miRNAs based on their area under the curve of the receiver operating characteristic curve (AUC-ROC) to separate the CAD/MI samples. We designed a two-layer architecture for sample classification, with the first layer isolating healthy samples from not-healthy ones, and the second layer classifying stable CAD and MI samples. We trained different machine learning models using both biomarker sets and evaluated their performance on a test set. We identified miR-21, miR-186, and miR-32 as the only miRNAs among the differentially expressed genes, and a set including miR-186, miR-21, miR-197, miR-29A, and miR-296 as the optimum set of miRNAs selected by their AUC-ROC. Both biomarker sets could distinguish healthy from not-healthy samples with complete accuracy. The best performance for the classification of CAD and MI was achieved with an SVM model trained using the biomarker set selected by AUC-ROC, with an AUC-ROC of 0.95, and an accuracy of 0.88. Our study demonstrated that miRNA signatures derived from peripheral blood could serve as valuable novel biomarkers for cardiovascular diseases.


output:
  rticles::springer_article
    
bibliography: [ForArticleRefrence.bib, discussion.bib, Introduction.bib, mir296.bib]
biblio-style: spmpsci
# bibstyle options spbasic(default), spphys, spmpsci
---

```{r setup, include=FALSE, cache.rebuild=TRUE}
knitr::opts_chunk$set(echo = TRUE)
if (!require("pacman")) install.packages("pacman")
pacman::p_load(knitr, captioner, bundesligR, stringr, kableExtra)
```

# Introduction

At present, cardiovascular diseases (CVDs) are the leading cause of human mortality with 32% of all global deaths. It is estimated that about 85% of CVDs mortality were diagnosed with myocardial infarction (MI) [@CVD]. MI is an acute coronary syndrome with sudden blockage and stenosis of the coronary artery, and subsequent myocardial ischemia, leading to extensive cardiomyocyte damage and necrosis [@Macro].

Over the last 50 years, numerous attempts have been collected to use biomarkers to facilitate diagnosis, assess risk, follow-up therapy, and determine therapeutic efficacy in CVDs candidates. Based on the released guidelines, cardiac troponins (cTns) are used as a highly-sensitive and accurate approach for the detection of MI. Despite the inherent advantages, the high-rate sensitivity of cTn-based assays has also led to more false positive results [@17], which do necessitate the advent and development of new modalities with pathological values. To improve diagnostic value upon existing MI biomarkers, the combination of complementary biological markers, such as microRNAs (miRNAs) and other genetic factors, is proposed. Previous researches support the notion that miRNAs exhibit the great potential to be used as alternative biomarkers in CVDs detection and follow-up [@miR]. It is suggested that miRNAs possess 18-22 nucleotides and can play a crucial role in the regulation of gene expression. Evidence point to the fact that miRNAs are involved in the pathogenesis of cardiac tissue injury and could be considered as theranostics targets in terms of CVDs. These elements can easily circulate in biofluids with both diagnostic as well as prognostic values [@m1]. Several biological activities such as angiogenesis, cardiomyocyte growth and contractility, lipid metabolism, plaque formation, and cardiac rhythm are regulated by miRNAs [@m2]. It is postulated that the function and diagnostic properties of miRNAs are beyond the myocardium in CVD patients. To be specific, the expression of miRNAs can vary in different biofluids and cell components such as serum and peripheral blood mononuclear cells (PBMCs) [@36].

PBMCs are a fraction of white blood cells (WBCs), including monocytes, lymphocytes, macrophages, and other cells belonging to the immune system [@PBMC-miR]. Emerging data have indicated that PBMCs can be used as a valid source of biomarkers for monitoring various pathological conditions. Of note, the alteration of mRNAs and miRNAs under pathological conditions gives us valuable information about different kinds of disorders. PBMCs could recapitulate the conditions of the target tissues, thus, providing a highly sensitive and specific source of biomarkers [@Meder6]. Commensurate with these conditions, these cells are repositories of dysregulated genes and miRNAs expression profiles in CVDs [@PBMC-miR; @Meder6].

In recent years, the advent and use of machine learning (ML) is an exciting prospect for advancing scientific discoveries. Although the concept of ML and its initial algorithms were conceived many years ago, recent improvements in computing power and access to vast amounts of data have shown that ML techniques outperform classical statistical methods in various fields. Furthermore, the progress made in omics technologies has enabled the analysis of massive and intricate biological data sets, consisting of hundreds to thousands of samples, which makes it possible for ML to extract valuable biological information from such data [@ML-Bio]. Therefore, ML offers novel techniques to integrate and analyze the various omics data enabling the discovery of de novo biomarkers. These biomarkers help us in accurate disease prediction, patient stratification, and finding new therapeutics [@ML].

In this study, we aimed to identify potential miRNA biomarkers for MI patients by combining and analyzing three different microarray datasets from PBMCs. It is suggested that the integration of omics data with bioinformatics and ML techniques could be a promising tool in the discovery of new and more accurate biomarkers for monitoring MI. Besides, this approach can deepen our vision into the underlying mechanisms of MI and aid in the development of valid diagnostic biomarkers, and patient stratification.

# Materials and Methods

## Microarray data collection

Microarray datasets were obtained from the Gene Expression Omnibus (GEO) database (<https://www.ncbi.nlm.nih.gov/geo/>). To obtain sufficient classification power between MI, healthy and stable coronary artery disease (CAD) samples, a relatively large sample size was required. Therefore, GSE59867 for MI and CAD samples, and GSE56609 and GSE54475 for healthy samples were selected. All samples were produced using Affymetrix Human Gene 1.0 ST Array (GPL6244) platform. Only healthy, stable CAD and early-stage MI samples were selected from these datasets for further analyses. The basic information for the three datasets evaluated in the current study is provided in Table \@ref(tab:datasets).

```{r, datasets, echo=FALSE}
df <- data.frame(Dataset = c("GSE59867", "GSE56609", "GSE54475"),
                 Platform = c("GPL6244", "GPL6244", "GPL6244"),
                 Healthy = c("-", 46, 5),
                 CAD = c(46, "-", "-"),
                 MI = c(111, "-", "-"),
                 Refrence = c("[@67]", "[@09]", "[@75]"))

kbl(df, format = "markdown", booktabs = T, align = "c", linesep = c(""), caption = "Basic information of the GEO microarray datasets.")
```

## Pre-processing

Raw data (CEL files) of all datasets were downloaded from the GEO and pre-processed using the fRMA package [@fRMA]. fRMA allowed to pre-process of individual microarray samples and combining them consistently for analysis. For each dataset, background correction was performed using the RMA algorithm and then it was quantile normalized based on the reference distribution. During summarization, batch effects were removed and variances of the gene expressions were estimated by taking into account these probe-specific effects. For those multiple probe sets matched to the identical gene, the mean log fold change was retained. Therefore, fRMA can be seen as a batch effect removal technique for different datasets that are produced by identical microarray platforms. Thus, to ensure batch effect removal, the principal component analysis and the relative log expression of train samples were plotted before and after fRMA [@BER].

## Differential expression analysis

The barcode algorithm proposed by McCall et al. [@Barcode] transformed the actual expression values into binary barcode values. Huge sets of samples were collected and normalized using fRMA for several platforms as well as for Affymetrix Human Gene 1.0 ST Array (GPL6244) platform. The distribution of the expressed and unexpressed observed intensities for each gene is estimated using these normalized sets. Genes were considered expressed (and their value coded to 1) or unexpressed (and their value coded to 0) according to the following equation:

$$
\hat{x}_{ij} = \left\{
  \begin{array}{ll}
    1 & \mbox{if } x_{ij} >= \mu^{ne} + C \times \sigma^{ne} \\
    0 & \mbox{otherwise}
  \end{array}
\right.
$$

where $x_{ij}$ is the normalized intensity of gene $i$ in sample $j$, $C$ is a user-defined parameter, $\sigma^{ne}$ is the standard deviation of the non-expressed distribution, and $\mu^{ne}$ is the mean of the non-expressed distribution. The barcode representation of a sample is a vector of ones and zeros denoting which genes are estimated to be expressed (ones) and unexpressed (zeros). The barcode algorithm was implemented by the barcode function in the R fRMA package, and the default value of $C$ was used.

To determine if the expressed ratios differed in the MI group versus the healthy control group, Fisher's exact test for individual genes was carried out upon the barcode values. Genes with a false discovery rate (FDR) of $< 0.05$, which was calculated through the Benjamini-Hochberg (BH) procedure to adjust for multiple testing issues, were considered as differentially expressed genes. The same procedures were conducted on CAD versus healthy controls as well as MI versus CAD group to find the DEGs between them.

## Functional and pathway enrichment analyses

Using the R clusterProfiler package, the Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway enrichment analysis and Gene Ontology (GO) functional annotation were carried out on the differentially expressed genes. The GO analysis included biological process (BP), cellular component (CC) and molecular function (MF) categories. An adjusted p-value of less than 0.05 was considered to indicate a statistically significant difference. Enrichments were conducted on the MI-healthy and CAD-healthy DEGs. In these analyses, all default parameters were used.

## ML procedure

The ML analysis was performed using Python software, ver. 3.9, Numpy [@numpy], pandas [@pandas], and Scikit-Learn packages [@scikit-learn]. Whenever hyper-tuning was needed, the scikit-opt package [@scikitopt] was used. In all ML analyses, the datasets were divided into train and test sets by a 0.7:0.3 ratio and all reported results are the average of 10-fold cross-validation.

Two different approaches were used for selecting miRNAs for model training. The first approach was using the miRNAs that are differentially expressed. In the second approach, miRNAs with individual area under the curve of the receiver operating characteristic (AUC-ROC) over 0.8 for separating MI from CAD were selected. Having the result of these two different approaches can provide an informative comparison between the predictive capabilities of sets of miRNAs selected with different logics.

### miRNAs in DEGs

In this approach, a two-layer architecture was deployed to the data to maximize the prediction values. The first layer predicted whether a sample is healthy or not, and the second layer separated MI from CAD in the samples which were predicted as not healthy in the first layer. To this end, a distinct ML model was trained for each layer. Since there is a limited number of miRNAs in DEGs, both layers were trained with all of them. For further comparison with the models' performance, the ROC curve of each miRNA for classifying healthy and not-healthy, as well as CAD and MI, were generated using a Logistic Regression model.

#### First layer for isolation of healthy and not-healthy samples:

An SVM model using RBF kernels was trained and hyper-tuned using all miRNAs in DEGs. To handle the severe imbalance in the number of samples (51 for the healthy and 157 for the not-healthy groups), the sample weight for the healthy and the not-healthy samples were set to 1 and 0.5, respectively.

#### Second layer for seperating MI and CAD samples:

For the sake of reaching the highest classification performance using the set of miRNAs, different models were investigated. To do so, SVM (with linear, polynomial, and RBF kernels), Logistic Regression (LR), Random Forests (RF), k-Nearest Neighbor (kNN), Gradient Boosting (GB), XGBoost (XGB) and Decision Tree (DT) models were trained. All models were trained with their pre-set parameters with 10-fold cross-validation. The criteria for choosing the best model were the highest accuracy and AUC-ROC on the test set. The best model was hyper-tuned with the scikit-opt package [@scikitopt] to get the best predictive performance.

### miRNAs with the highest AUC-ROC

Like the previous approach, a two-layers strategy was conducted. The first layer classified samples into healthy and not-healthy, and the separated MI and CAD samples. However, to keep the number of miRNAs as low as possible miRNAs were selected from the second layer, and then their performance wase evaluated in the first layer. AUC-ROC of all miRNAs for classifying MI and CAD samples were calculated and the miRNAs with the AUC-ROC over 0.8 were selected. The ROC curves for each selected miRNA for separating healthy samples from not-healthy and MI from CAD samples were also plotted for further comparison.

#### First layer for the detection of healthy and not-healthy samples:

An SVM model with an RBF kernel was trained using the selected set of miRNAs. Additionally, the model was hyper-tuned to find the hyper-parameters for the highest AUC-ROC and accuracy. The ROC curve and confusion matrix for the best model were reported.

#### Second layer for separating MI and CAD:

The selected miRNAs set was used to train different algorithms to find the best model. Similar to the previous approach, SVM (with linear, polynomial, and RBF kernels), LR, RF, kNN, GB, XGB, and DT were trained. All models were trained with their pre-set parameters using 10-fold cross-validation. The models with the highest AUC-ROC and accuracy on the test set were selected and hyper-tuned using the scikit-opt package [@scikitopt]. The ROC curve and confusion matrix for the best model were reported.

# Results

## Pre-processing

The PCA plots of the training samples are shown in Figure \@ref(fig:PCA)A and B. As shown, healthy samples were separated from CAD or MI samples in primary data and also after conducting fRMA. In the RLE plot, there was a distinct difference between dataset means for all samples before conducting fRMA (Figure \@ref(fig:PCA)C). All datasets were rearranged around 0 in the RLE plot after conducting fRMA (Figure \@ref(fig:PCA)D). Moreover, there was a clear change in inter-quantile distances, but the values still were over 0.1.

```{r PCA, echo=F, out.width="100%", fig.cap="PCA and RLE plot for all samples before and after fRMA."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/DEG and DEM/Pics/PCA and RLE of trains.pdf")
```

## Differential expression analysis

```{r , DEGstab, echo=F}
M_H <- read.table("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/DEG and DEM/Fisher results/DEGs MI-Healthy.txt", check.names = F)

M_C <- read.table("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/DEG and DEM/Fisher results/DEGs MI-CAD.txt", check.names = F)
  
C_H <- read.table("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/DEG and DEM/Fisher results/DEGs CAD-Healthy.txt", check.names = F)

df <- t(data.frame(`MI vs. Healthy` = c(dim(M_H)[1], sum(M_H$logFC > 0), sum(M_H$logFC < 0), "hsa-miR-186, miR-21, miR-32"),
       `CAD vs. Healthy` =  c(dim(C_H)[1], sum(C_H$logFC > 0), sum(C_H$logFC < 0), "hsa-miR-186, miR-21, miR-32"), 
       `MI vs. CAD` = c(dim(M_C)[1], sum(M_C$logFC > 0),  sum(M_C$logFC < 0), "hsa-miR-186")))

colnames(df) <- c("Total DEGs",  "Up-regulated DEGs", "Down-regulated DEGs", "miRNAs")
rownames(df) <- c("MI vs. Healthy", "CAD vs. Healthy", "MI vs. CAD")

kbl(df, booktabs = T, align = "c", linesep = c(""), caption = "Total, up-, and down-regulated DEGs and differentially expressed miRNAs.")%>% 
  kable_styling(full_width = T) %>%
  column_spec(1, width = "3cm") %>%
  column_spec(5, width = "5cm") %>%
  column_spec(4, width = "2.5cm") %>%
  column_spec(3, width = "2.5cm")
```

According to the cutoff criterion of $FDR < 0.05$, there were `r dim(M_H)[1]` DEGs between the MI and the healthy samples. Among them, `r sum(M_H$logFC > 0)` were up-regulated, and `r sum(M_H$logFC < 0)` were down-regulated in MI compared to the healthy controls. In CAD and healthy groups comparison, we found `r dim(C_H)[1]` DEGs, of which `r sum(C_H$logFC > 0)` and and `r sum(C_H$logFC < 0)` DEGs were up- and down-regulated, respectively in CAD samples. In the MI and CAD groups, the number of DEGs was `r dim(M_C)[1]`, and the number of up- and down-regulated genes in MI samples were `r sum(M_C$logFC > 0)` and `r sum(M_C$logFC < 0)`, respectively in comparison with CAD samples. These data are summarized in Table \@ref(tab:DEGstab).

```{r venn, echo=F, out.width="45%", fig.align='center', fig.cap="Venn diagram for DEGs in CAD/Healthy, MI/Healthy, and MI/CAD comparison."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/DEG and DEM/Pics/venny.png")
```

The Venn diagram in Figure \@ref(fig:venn) shows that CAD and MI samples share a majority of their DEGs. From 860 DEGs of MI/healthy and 670 DEGs of CAD/healthy, 531 genes were common which is 62% of MI/healthy DEGs and 79% of CAD/healthy DEGs.

## Gene ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) enrichment analyses of the DEGs.

To explore the biological classification of the DEGs, we performed GO and KEGG pathway enrichment analyses on MI-healthy and CAD-healthy DEGs. In MI versus healthy samples, GO enrichment analysis in the BP category, suggested that the DEGs were enriched in "immune response-regulating signaling pathway", "lymphocyte differentiation", "immune response-regulating cell surface receptor signaling pathway", and "leukocyte activation involved in immune response" (Figure \@ref(fig:MIHEnrich)A). In the CC category, the DEGs were enriched in "secretory granule membrane", "azurophil granule", "ficolin-1-rich granule", "tertiary granule", and "ficolin-1-rich granule membrane" (Figure \@ref(fig:MIHEnrich)B). In the MF category, the DEGs were involved in "cadherin binding" and "MHC class I protein binding" (Figure \@ref(fig:MIHEnrich)C). KEGG pathway analysis indicated that the DEGs were related to the following pathways: "Chemokine signaling pathway", "Lipid and atherosclerosis", and "Hematopoietic cell lineage" (Figure \@ref(fig:MIHEnrich)D).

```{r MIHEnrich, echo=F, out.width="95%", fig.cap="Gene Ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) pathways enriched with the MI and healthy DEGs. (A) Biological process terms. (B) Cellular component terms. (C) Molecular function terms. (D) KEGG analysis."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/DEG and DEM/Pics/Enrichment MI-Healthy.pdf")
```

The enrichment results for DEGs of CAD versus healthy samples were as follows. In the BP category, GO enrichment suggested that the DEGs were enriched in "positive regulation of defense response", "positive regulation of innate immune response", "mononuclear cell differentiation", and "positive regulation of response to external stimulus" (Figure \@ref(fig:CADHEnrich)A). In the CC category, the DEGs were enriched in "azurophil granule", "ficolin-1-rich granule", and "ficolin-1-rich granule membrane" (Figure \@ref(fig:CADHEnrich)B). In the MF category, the DEGs were involved in "lipoprotein particle receptor binding" and "NF-$\kappa$B binding" (Figure \@ref(fig:CADHEnrich)C). KEGG pathway analysis indicated that the DEGs were related to the following pathways: "Chemokine signaling pathway", "Lipid and atherosclerosis", and "Hematopoietic cell lineage" (Figure \@ref(fig:CADHEnrich)D).

```{r CADHEnrich, echo=F, out.width="95%", fig.cap="Gene Ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) pathways enriched with the CAD and healthy DEGs. (A) Biological process terms. (B) Cellular component terms. (C) Molecular function terms. (D) KEGG analysis."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/DEG and DEM/Pics/Enrichment CAD-Healthy.pdf")
```

## Machine Learning

### miRNAs in DEGs

Among all DEGs, just miR-186, miR-32, and miR-21 were detected as differentially expressed miRNAs. The expression profile of these three miRNAs is presented in Figure \@ref(fig:Expall). Additionally, The ROC curves of each miRNA for each layer are presented in Figure \@ref(fig:miRROC). Using the logistic regression model the AUC-ROC of miR-21, miR-32, and miR-186 for separating healthy and not-healthy samples was 0.98, 0.99, and 0.90 respectively (Figure \@ref(fig:miRROC)A). Besides, the accuracy of each miRNA for classifying the samples into healthy and not-healthy groups on the test set was 0.92, 0.98, and 0.89 for miR-21, miR-32, and miR-186, respectively. Moreover, the ROC curve of each miRNA for classifying MI and CAD samples was presented in Figure \@ref(fig:miRROC)B. For miR-21, miR-32, and miR-186, the AUC-ROC and accuracy on the test set were 0.85; 0.70; and 0.86, and 0.78; 0.67; and 0.74, respectively.

```{r mirExptable, echo=FALSE}

# data
df <- read.table("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/DEG and DEM/Pics/microRNAs logFC.txt", sep = "\t", check.names = F, header = T)

kbl(df, booktabs = T, align = "c",linesep = c(""), digits = c(1, 60), caption = "Target miRNAs log fold-change and adjusted p-values for CAD samples relative to healthy, MI samples relative to healthy, and MI samples relative to CAD.") %>%
  kable_styling(full_width = T) %>%
  add_header_above(c(" " = 1, "CAD/Healthy" = 2, "MI/Healthy" = 2, "MI/CAD" = 2))
```


```{r Expall, echo=F, out.width="90%", fig.align='center', fig.cap="Expression profile of all miRNAs in two approach in different sample classes."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/Expressionforall.pdf")
```


```{r miRROC, echo=F, out.width="95%", fig.align='center', fig.cap="ROC curve for single miRNAs on test set classification for (A) healthy and not-healthy samples and (B) CAD and MI samples."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/miRs ROCs.pdf")
```

#### First layer for healthy not-healthy isolation:

Although single miRNAs had acceptable performance, their predictive value could be improved even further by using them as a set. The ROC curve for the SVM model with an RBF kernel trained with all three miRNAs is presented in Figure \@ref(fig:DEMROC)A. The model had a better performance in classification than single miRNAs.The AUC-ROC for the model is 1, and its accuracy on the test set was also 1. In Figure \@ref(fig:DEMCF)A, the confusion matrix for the model is presented.

```{r DEMROC, echo=F, out.width="95%", fig.align='center', fig.cap="ROC curve for miRNAs in DEGs on test set classification for (A) healthy and not-healthy samples and (B) CAD and MI samples."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/DEMs ROCs h not h cad mi.pdf")
```

```{r DEMCF, echo=F, out.width="90%", fig.align='center', fig.cap="Confusion matrix on the test set for (A) An SVM model with RBF kernel for healthy and not-healthy and (B) An SVM model with linear kernel for CAD and MI samples classification."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/DEMs Conf. matrix.pdf")
```

#### Second layer for separating MI samples from CAD:

Different models were trained using expression values for three differentially miRNAs. The models' 10-fold cross-validated AUC-ROC and accuracy on the test set are reported in Figure \@ref(fig:DEMmodels). The best model from both AUC-ROC and accuracy point-of-view was the SVM model with linear kernel. The AUC-ROC and accuracy for this model with its pre-set values were 0.93 and 0.82 respectively. The model was hyper-tuned for C and gamma hyper-parameters, and therefore the model showed better performance. The ROC curve of the hyper-tuned model is presented in Figure \@ref(fig:DEMROC)B. For this model the AUC-ROC reached 0.95 and the accuracy improved to 0.85 (Table \@ref(tab:DEGsML)). Moreover, the sensitivity and specificity for the model on the test set were 0.91 and 0.71 respectively. The confusion matrix for the hyper-tuned model is illustrated in Figure \@ref(fig:DEMCF)B.

```{r DEMmodels, echo=F, out.width="65%", fig.align='center', fig.cap="Area under the receiver operating characteristic curve and accuracy of different models trained with three miRNAs in DEGs on the test set."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/Models DEGs mirs.pdf")
```

```{r, DEGsML, echo=FALSE}

# data
df <- read.table("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/DEGsMICAD.txt", sep = "\t", check.names = F, header = T)

kbl(df, booktabs = T, align = "c",linesep = c(""), caption = "AUC-ROC and accuracy for SVM with the linear kernel as the best model trained with differentially expressed miRNAs on the train and test set before and after hyper-tuning") %>%
  kable_styling(full_width = T) %>%
  add_header_above(c(" " = 2, "Pre-set parameters" = 2, "Hyper-tuned" = 2))
```

### AUC-ROC approach

After calculating the AUC-ROC for each miRNA for the classification of MI and CAD samples, the miRNAs with AUC-ROC over 0.8 were selected. The selected miRNAs were miR-29a, miR-197, miR-186, miR-21, and miR-296. The expression values of these miRNAs in healthy, CAD, and MI samples are presented in Figure \@ref(fig:Expall). The ROC curve of the selected miRNAs for both layers are illustrated in Figure \@ref(fig:miRROC).

#### First layer for isolation of healthy and not-healthy samples

Using the selected set, an SVM model with an RBF kernel was trained to separate healthy from not-healthy samples. The ROC curve for the model is presented in Figure \@ref(fig:AUCROC)A and the confusion matrix is illustrated in Figure \@ref(fig:AUCCF)A. Both AUC-ROCand accuracy for the model on the test set were equal to 1.

```{r AUCROC, echo=F, out.width="95%", fig.align='center', fig.cap="ROC curve for the set of miRNAs selected by AUC-ROC on test set classification. (A) SVM with RBF kernel for healthy and not-healthy samples classification. (B) Logistic regression model for CAD and MI samples classification. (C) SVM with polynomial kernel for CAD and MI samples classification. "}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/AUC best model ROCs.pdf")
```

```{r AUCCF, echo=F, out.width="90%", fig.align='center', fig.cap="Confusion matrix on the test set for (A) SVM with RBF kernel for healthy and not-healthy samples classification. (B) Logistic regression model for CAD and MI samples classification. (C) SVM with polynomial kernel for CAD and MI samples classification."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/AUC Conf. matrix.pdf")
```

#### Second layer for isolation of MI samples from CAD samples

To find the best model for training the best set, different models were trained using their pre-set values. Their AUC-ROC and accuracy results on the test set are presented in Figure \@ref(fig:AUCmodels). The best model from the AUC-ROC point-of-view was the SVM with linear kernel and from the accuracy point-of-view, it was the SVM model with an RBF kernel. For the SVM-linear model the AUC-ROC and accuracy were 0.93 and 0.82, respectively; and for the SVM-RBF, the values were 0.92 and 0.84, respectively. Both models were hyper-tuned and the ROC curve for their best performance is presented in Figure \@ref(fig:AUCROC)B and C. The AUC-ROC and accuracy for the SVM-linear model modified to 0.92 and 0.88, respectively. For the SVM-RBF, these values increased to 0.96 and 0.94, respectively (Table \@ref(tab:AUCML)). The sensitivity for the SVM-linear and SVM-RBF models were 0.91 and 0.97, respectively; and the specificity for them was 0.79 and 0.86, respectively. The confusion matrix for both models is illustrated in Figure \@ref(fig:AUCCF)B and C.

```{r AUCmodels, echo=F, out.width="65%", fig.align='center', fig.cap="Area under the receiver operating characteristic curve and accuracy of different models trained with AUC-selected miRNAs."}
knitr::include_graphics("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/Models AUC 5 mirs CV.pdf")
```

```{r AUCML, echo=FALSE}

# data
df <- read.table("/Users/mehrdad/Projects/TBZ.Med/PhD. Project/Bioinformatics and ML/Article Materials/ML/Pics/AUCMICAD.txt", sep = "\t", check.names = F, header = T)

kbl(df, booktabs = T, align = "c",linesep = c(""), caption = "AUC-ROC and accuracy for SVM with the linear kernel as the best model trained with miRNAs selected based on their individual AUC-ROC on the train and test set before and after hyper-tuning") %>%
  kable_styling(full_width = T) %>%
  add_header_above(c(" " = 2, "Pre-set parameters" = 2, "Hyper-tuned" = 2))
```

# Discussion

The prevalence of MI can lead to high-rate mortality in the clinical setting. However, early diagnosis and application of suitable treatment protocols can reduce mortality and improve AMI prognosis [@CVD; @17; @18]. Studies have suggested that changes in miRNA expression may play a significant role in the progression of MI and the subsequent remodeling [@23]. It is believed that the expression of miRNAs is altered during the various biological processes correlated with MI within the myocardium or other related tissues [@24]. Although several research has been concentrated on examining free circulating miRNAs in the serum samples for the detection of cardiac tissue injuries [@11-15], more information is needed to fully comprehend the miRNAs found in different blood sub-components like plasma, platelets, and PBMCs. Based on previous data, PBMCs are critically involved in plaque destabilization and rupture as well as early inflammatory responses during MI [@Meder6; @Meder23]. Moreover, PBMCs have specific miRNA profile that is altered under certain pathological conditions which are great candidates as disease biomarkers [@Meder6].

PBMCs can respond to several insulting conditions such as MI in the least possible time with prominent changes in their miRNA profile [@Meder6]. Considering the regulatory roles, subtle changes in the transcription of miRNAs can be monitored even before alteration in the levels of mRNAs and proteins [@miR]. These features make the miRNAs an early-stage valid diagnostic tool for the detection of minor and major cell injuries. To date, few studies have been performed to compare the miRNA profiles in PBMCs belonging to acute MI patients and other CADs and healthy samples to find a robust set of identical miRNAs to differentiate these pathological conditions.

In this study, we combined three GEO datasets of healthy, CAD, and MI samples. Having these samples set alongside bioinformatics analysis and ML means, it is possible to identify potential biomarker sets and also effective therapeutic targets. The results of the DEG analysis (Table \@ref(tab:DEGstab) and Figure \@ref(fig:venn)) are proof of the close relationship between the MI and CAD samples. Interestingly, functional enrichment analysis demonstrated that DEGs in both healthy/CAD and healthy/MI were strongly correlated to immune cell response which is a major cellular part of PBMCs. Here, two different sets of miRNAs were used as biomarker sets for sample classification. miR-21; miR-32; and miR-186 were selected as differentially expressed miRNAs, and miR-186, miR-21; miR-29a; miR-197; and miR-296 were selected according to their AUC-ROC values. As shown in Figure \@ref(fig:miRROC), all miRNAs selected with both approaches had AUC-ROC over 0.9 for the isolation of healthy and not-healthy samples except for miR-296 and miR-29a. Data confirmed that the real challenge is to classify CAD and MI samples because of close overlap. Of 6 miRNAs under investigation in both approaches except for miR-32, all miRNAs had an AUC-ROC over 0.8 for the discrimination of CAD and MI samples. Besides, the high AUC-ROC values of miRNAs confirms their high potential as biomarkers.

ML models when trained with miRNA sets selected by both DEG and AUC-ROC approaches showed better performance in the classification than each miRNA. To avoid unwanted complexity and poor predictive values, a two-layer architecture was also designed. The first layer was for the discrimination of healthy from not-healthy samples, and the second layer separated the CAD from MI candidates. As expected, in both approaches a hyper-tuned SVM model could flawlessly separates healthy from not-healthy samples using distinct miRNAs sets. The ML models were also capable of effectively separating CAD from MI patients. Although both miRNA sets had nearly the same AUC-ROC with their best model, the accuracy, sensitivity, and specificity were different. The model trained with AUC-selected miRNAs had better performance in all predictive values.

Numerous studies have reported different biological processes can affect the expression of miRNAs in PBMCs. However, there are still controversies regarding the exact role of miRNAs in the function of immune cells and the correlation of specific pathological conditions with miRNA profiles. Several studies have proved the activation of specific miRNA types in PBMCs under cardiovascular events [@21; @186; @32-2; @29aa; @197]. For instance, there is evidence that the elevation of miR-186 suppresses the expression of Cystathionine-$\gamma$-lyase, leading to the subsequent secretion of pro-inflammatory cytokines and cellular lipid accumulation. Besides, macrophage-derived miR-186 may promote atherosclerotic plaques [@186]. In line with this claim, we found that miR-186 is up-regulated in both CAD and MI candidates related to control counterparts. Surprisingly, the obtained data indicated that the expression of miR-186 is higher in CAD patients in comparison to MI (Figure \@ref(fig:DEMexp)), To be specific, miR-186 is the only differentially expressed miRNA between CAD and MI, with a clear up-regulation in CAD, indicating its main role in the promotion of atherosclerosis.

As mentioned before, miR-21 was also up-regulated in both MI and CAD patients in comparison to healthy controls. Moreover, the expression value of miR-21 was significantly higher in MI than that of the CAD group (Table \@ref(tab:mirExptable)). It is thought that the up-regulation of miRNA-21 in PBMCs is a compensatory reaction to reduce T$_{reg}$ lymphocyte number in response to the reduction of TGF$\beta1$ secretion into the plasma through a TGF$\beta1$/smad-independent pathway. In line with previous and present data, miR-21 can modulate the activity of PBMCs following the occurrence of cardiovascular diseases [@21].

Recent data have supported the elevation of miR-32 in CAD patients with the calcification of coronary artery. Of note, miR-32 stimulates the calcification of mouse vascular smooth muscle through the regulation of bone morphogenetic protein-1, runt-related transcription factor-2 (RUNX2), osteopontin, and bone-specific phosphoprotein matrix GLA protein [@32-2]. Likewise, there are some reports associated with the activity of PBMC miR-32 under several pathologies [@magic; @32]. The exact role of PBMC miR-32 after cardiovascular events remained to be elucidated.

Molecular analyses have indicated the regulatory role of miRNAs selected by the AUC-ROC approach in PBMCs after a cardiovascular event. Two common miRNAs in DEGs and AUC-ROC approachs, miR-21 and miR-186, were covered in DEGs miRNA set. Based on numerous reports mir-29a can be activated in different diseases [@29aa].Data analysis indicated that miR-29a is significantly up-regulated in CAD patients in comparison to healthy and MI groups (Table \@ref(tab:mirExptable)). Increased miR-29a is associated with the progression of atherosclerosis, and the combination of miR-29a and ox-LDL was offered as a valid biomarker set for paraclinical classification [@29a]. However, the role of miR-29a in the function of PBMCs in CAD patients has not been completely examined.

Data indicated that miR-197 is also significantly up-regulated in both CAD/healthy and MI/healthy groups. In previous studies, it has been shown that miRNA-197 may play a critical role in regulating the anti-inflammatory response of IL-35 by affecting pro/ anti-inflammatory cytokine secretion, M1/M2 macrophage ratio, T$_{reg}$ lymphocyte proliferation, and T cell suppression suggesting the potential diagnostic role of miR-197 in adverse cardiovascular events [@197]. 

The findings of this study demonstrate the potential of miR-296 as a biomarker with high discriminatory power for distinguishing between samples from individuals with MI and CAD. MiR-296 has been identified as a key regulator in the development and advancement of atherosclerosis by controlling the expression of target genes associated with various biological processes including angiogenesis, inflammation, cholesterol metabolism, hypertension, cellular proliferation, and apoptosis [@296]. In a study it has been shown that miR-296 expression levels are significantly increased in the PBMCs of CAD patients compared to healthy controls, suggesting its involvement in regulating pro-inflammatory cytokines like IL-6 and TNF-a [@296a]. These findings suggest that miR-296 may have a significant impact on the pathogenesis of atherosclerosis and could potentially serve as a diagnostic biomarker for CAD or MI.

# Conclusion

In summary, we derived a set of miRNA biomarkers by comparing MI samples to both healthy and CAD samples. We found that the SVM model performed best in both the first layer, which separated healthy and not-healthy samples, and the second layer, which classified MI/CAD samples. The set of miRNAs selected based on their AUC-ROC values had better performance in the second layer. Overall, our two-layer structure achieved a accuracy of 0.96. This demonstrates the potential for combining bioinformatics and machine learning techniques to identify novel biomarkers and gain a deeper understanding of myocardial infarction.


# References


\begin{align}
a^2+b^2=c^2
\end{align}


